airflow/utils/log/es_task_handler.py

```diff
@@ -141,10 +141,10 @@ def _read(self, ti, try_number, metadata=None):
         if 'last_log_timestamp' in metadata:
             last_log_ts = timezone.parse(metadata['last_log_timestamp'])
             if cur_ts.diff(last_log_ts).in_minutes() >= 5 or 'max_offset' in metadata \
-                    and offset >= metadata['max_offset']:
+                    and int(offset) >= int(metadata['max_offset']):
                 metadata['end_of_log'] = True
 
-        if offset != next_offset or 'last_log_timestamp' not in metadata:
+        if int(offset) != int(next_offset) or 'last_log_timestamp' not in metadata:
```

Comment:
A string comparison is okay here isn't it?

Reply:
it tries to compare an int to a string in some cases

Reply:
I want to ultimately get to the bottom of it, this is kind of just covering our bases I guess

Reply:
Ah okay. The "proper" fix for this might be to convert the offsets to an int inside the web view, so that the log handler only ever has to deal with ints.

Reply:
Ok, thank you. I will make a patch like that for upstream.